GVFS/GVFS.Platform.Mac/MacFileSystem.cs

```diff
@@ -0,0 +1,157 @@
+﻿using GVFS.Common;
+using GVFS.Platform.POSIX;
+using System;
+using System.Runtime.InteropServices;
+
+namespace GVFS.Platform.Mac
+{
+    public class MacFileSystem : POSIXFileSystem
+    {
+        public override void ChangeMode(string path, ushort mode)
+        {
+            Chmod(path, mode);

```

Comment:
Why does this not work on Linux?  Is `mode_t` something other than `ushort`?

Reply:
Yes, it's a `uint` -- so the `Chmod` DLL import wrapper is [different on Linux](https://github.com/github/VFSForGit/blob/6e457bc631f43d69681cb7693df0e36bdec3f7fb/GVFS/GVFS.Platform.Linux/LinuxFileSystem.cs#L33), and hence `ChangeMode()` ends up having to be in the OS-specific class rather than the base POSIX class, even though it looks [the same](https://github.com/github/VFSForGit/blob/6e457bc631f43d69681cb7693df0e36bdec3f7fb/GVFS/GVFS.Platform.Linux/LinuxFileSystem.cs#L12).  You can see these Linux variants in #1125.

Reply:
On Linux, it's a 32-bit `int` (assuming we're using fairly generic hardware and compiler, of course):
```
$ gcc -E -xc -include 'sys/types.h' - </dev/null | grep mode_t
typedef unsigned int __mode_t;
typedef __mode_t mode_t;
```

On Mac:
```
gcc -E -xc -include 'sys/types.h' - </dev/null | grep mode_t | grep ^typedef
typedef __uint16_t __darwin_mode_t;
typedef __darwin_mode_t mode_t;
```